File: LogMyPos-Backend/Model/Dto/User/UserResponse.cs
using System.Text.Json.Serialization;

namespace LogMyPos_Backend.Model.Dto.User;

public sealed class UserResponse {
	public required Guid Id { get; init; }
	public required string Username { get; init; }
	public required string Email { get; init; }
	[JsonConverter(typeof(JsonStringEnumConverter))]
	public required Model.User.UserRole Role { get; init; }
	public required DateTime CreatedAt { get; init; }
	public required DateTime UpdatedAt { get; init; }
}

File: src/main/kotlin/dev/pandasystems/logmyposbackend/dto/UserResponse.kt
package dev.pandasystems.logmyposbackend.dto

import dev.pandasystems.logmyposbackend.model.User
import java.util.UUID

data class UserResponse(
	val id: UUID,
	val username: String,
	val email: String,
	val role: User.Role
)

fun User.toResponse() = UserResponse(
	id = this.id!!,
	username = this.username,
	email = this.email,
	role = this.role
)